uniffi_bindgen/src/interface/function.rs

```diff
@@ -249,6 +249,42 @@ impl APIConverter<Argument> for weedle::argument::SingleArgument<'_> {
     }
 }
 
+/// Implemented by function-like types (Function, Method, Constructor)
+pub trait Callable {
```

Comment:
out of curiosity, why is the `arguments` API return a vec of references where the other two return owned `Type`?

Is it because it was hard to have then all return references without overcomplicating the code?

Just curious!

Reply:
Additionally, would we like to add a `name` function too?

Reply:
> out of curiosity, why is the `arguments` API return a vec of references where the other two return owned `Type`?

I think that was copied from the #1469 implementation.   I didn't put a lot of thought into it, but that seemed like the path of least resistance to me.  In general, it seems like we tend to clone Type rather than try to manage a reference to it.  I'm not sure if that's the best policy going forward or not.

> Additionally, would we like to add a name function too?

Maybe?  I'm not totally sure though, there could be callables that don't have a name.  For example a destructor.  Also, the primary constructor currently has a name field, but it's just hard coded to "new".  That said, I'm not really against it I just didn't put the time into defining it.